api/types/server.go

```diff
@@ -492,20 +496,24 @@ func (s *ServerV2) CheckAndSetDefaults() error {
 // MatchSearch goes through select field values and tries to
 // match against the list of search values.
 func (s *ServerV2) MatchSearch(values []string) bool {
-	var fieldVals []string
-	var custom func(val string) bool
+	if s.GetKind() != KindNode {
+		return false
+	}
+
+	custom := func(val string) bool {
+		return s.GetUseTunnel() && strings.EqualFold(val, "tunnel")
+	}
```

Comment:
nit: I believe this is actually adding an additional allocation in comparison to the old behavior of only creating the closure when `s.GetUseTunnel()` is `true`, as this new closure captures a value from its surroundings (the `s` pointer), whereas the previous one did not (and therefore should be optimized into being a normal function pointer rather than a  true closure).